src/network/protocol/messages/lib.zig

```diff
@@ -109,3 +110,21 @@ pub const Message = union(MessageTypes) {
         };
     }
 };
+
+pub fn genericChecksum(m: anytype, hasPayload: bool) [4]u8 {
+    comptime {
+        if (!std.meta.hasFn(@TypeOf(m), "serializeToWriter")) @compileError("Expects m to have fn 'serializeToWriter'.");
+    }
+    if (!hasPayload) {
+        return [4]u8{ 0x5d, 0xf6, 0xe0, 0xe2 };
+    }
```

Comment:
don't handle the no payload in the generic method. Just don't call `genericChecksum` for those without payload and `return [4]u8{ 0x5d, 0xf6, 0xe0, 0xe2 }` directly